resources/Shaders/TexturePixel.hlsl

```diff
@@ -0,0 +1,12 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See License.md in the repository root for more information.
+// Originally taken from the TerraFX repository. Licenced under the MIT Licence (MIT). See Licence.md in https://github.com/terrafx/terrafx for more information.
+
+#include "TextureTypes.hlsl"
```

Comment:
Curious, do we want to add these to be formatted by clang-format if possible? Or no?

Reply:
If clang-format can support it then a different ruleset for HLSL would be pretty neat, but I don't think it'd be the same ruleset as the C++ code.

Reply:
Fair enough - clang-format is pretty touchy atm so let's review it later 😄